src/dsp_tools/xmllib/models/root.py

```diff
@@ -5,51 +5,178 @@
 
 # ruff: noqa: D101
 
-XML_NAMESPACE_MAP = {None: "https://dasch.swiss/schema", "xsi": "http://www.w3.org/2001/XMLSchema-instance"}
-DASCH_SCHEMA = "{https://dasch.swiss/schema}"
 
 AnyResource: TypeAlias = Union[
     Resource, AnnotationResource, RegionResource, LinkResource, VideoSegmentResource, AudioSegmentResource
 ]
 
 
+PRE_DEFINED_LICENSES = [
+    License("CC_BY", "CC BY 4.0", "https://creativecommons.org/licenses/by/4.0/"),
+    License("CC_BY_SA", "CC BY-SA 4.0", "https://creativecommons.org/licenses/by-sa/4.0/"),
+    License("CC_BY_NC", "CC BY-NC 4.0", "https://creativecommons.org/licenses/by-nc/4.0/"),
+    License("CC_BY_NC_SA", "CC BY-NC-SA 4.0", "https://creativecommons.org/licenses/by-nc-sa/4.0/"),
+    License("CC_BY_ND", "CC BY-ND 4.0", "https://creativecommons.org/licenses/by-nd/4.0/"),
+    License("CC_BY_NC_ND", "CC BY-NC-ND 4.0", "https://creativecommons.org/licenses/by-nc-nd/4.0/"),
+    License("CC0", "CC0 1.0", "https://creativecommons.org/publicdomain/zero/1.0/"),
+    License("unknown", "Copyright Not Evaluated", "https://rightsstatements.org/page/CNE/1.0/"),
+]
+
+
 @dataclass
 class XMLRoot:
     shortcode: str
     default_ontology: str
+    copyright_attributions: CopyrightAttributions
+    licenses: Licenses
     resources: list[AnyResource] = field(default_factory=list)
 
     @staticmethod
     def create_new(shortcode: str, default_ontology: str) -> XMLRoot:
         """
         Create a new XML root, for one file.
 
+        The following elements are added by default:
+            - `<permissions>` (with default permissions, see [Configuration options](https://docs.dasch.swiss/latest/DSP-TOOLS/xmllib-api-reference/config-options/#xmllib.models.config_options.Permissions) for details)
+            - `<copyright-attributions>` (empty)
+            - `<licenses>` (with default licenses, see [Configuration options](https://docs.dasch.swiss/latest/DSP-TOOLS/xmllib-api-reference/config-options/#xmllib.models.config_options.PredefinedLicenses) for details)
+
         Args:
             shortcode: project shortcode
             default_ontology: name of the default ontology
 
         Returns:
             Instance of `XMLRoot`
+        """  # noqa: E501 (Line too long)
+        return XMLRoot(
+            shortcode=shortcode,
+            default_ontology=default_ontology,
+            copyright_attributions=CopyrightAttributions(),
+            licenses=Licenses(PRE_DEFINED_LICENSES),
+        )
+
+    def add_license(self, id_: str, text: str, uri: Any = None) -> XMLRoot:
+        """
+        Add a new license element.
+        Please note that the id must be unique.
+
+        Args:
+            id_: ID which is referenced in the attributes of the XML
+            text: Text that should be displayed in the APP.
+            uri: Optional URI liking to the license documentation.
+                A `pd.isna()` check is done before adding the URI, therefore any value is permissible.
```

Comment:
My motivation was this:
- we never implicitly do not add something if the input is not what we expect
- I don't want to warn if it is not a string, because not having a url is valid
- I wanted to tell the user that it will not crash if the url is not a string
- I want to tell them than we will just not add it
- I want them to know what we consider to be valid.

We can also say that we check if it is a string and otherwise not add it...

Reply:
> I wanted to tell the user that it will not crash if the url is not a string

Hmm, so an integer would also be added as URI. I don't think that's a good idea. I suggest something along these lines:

```
        new_uri = None
        if not pd.isna(uri):
            if not is_string_like(uri):
                warnings.warn(f"URI must be a string. Got {type(uri)} instead.")
            new_uri = uri
```